### PR TITLE
Continuous distribution

### PR DESCRIPTION
**Pull request by jeandemeusy**
This PR introduce the last major iteration of CT. The biggest change is the introduction of continuous distribution.

## Main change
Compared to the previous implementation where messages were distributed every X minutes to all peers "at once", now messages are constantly sent to peers, with a delay between two messages specific to every peer. 
This is achieved by instantiating an `asyncio.Queue` object which is continuously filled by the `Peer` objects. The consumers of this queue are the HOPRd nodes used by CT. The content of an item in the queue is solely peer id's. This is the only necessary information from a peer to relay a message, as the sender/receiver is a random node, and message content plays no role in the distribution itself.

The way distribution records are stored in the database also changed. Before, it was possible to update the database with a single entry per distribution, after the distribution. Now in continuous mode, storing every single message wouldn't be possible as it would fill the database too fast and connection would slow things down to much. 
Instead, two threshold are now used: `min_value` and `timeout`. Basically, for every peer, an entry is added to the database either once the distributed messages are worth the `min_value` wxHOPR, or when the last db entry is older than the `timeout`.
This ensures that a single entry in the database covers a significant amount of tickets, or that it hasn't been too long since last update.

## Other changes
- no more retries, as in continuous distribution only a single ticket would be lost "per distribution"
- the same tag is used for all peers (`0x1245`). Instead of differentiating messages by inbox tag, they are linked to the relayer by the content itself
- keep `Peer` instances iterations after iterations: this is needed as the `Peer` instance handles the relay request themself, which is task running forever. Having the instances kept overtime without recreating them avoids duplicates  relay request tasks
- better subgraph handling
- small test coverage increase
- removal of unused prometheus metrics, merging of remaining ones, and renaming with `ct_` prefix
- database connection is kept alive for the whole execution, instead of re-initiating it for each db operation

## Summary by CodeRabbit

- **New Features**
	- Introduced a new message queue and formatting system for enhanced message handling.
	- Added new asynchronous methods in various components for improved task management.
	- Enhanced API functionality with updated logging and method signatures for better clarity.
	- Significant updates to configuration settings, improving core functionalities and economic model parameters.

- **Bug Fixes**
	- Refactored several components to improve usability and readability, including the `ChannelStatus` and `AsyncLoop` classes.

- **Documentation**
	- Added a template for environment variable configuration for consistent setup across environments.

- **Chores**
	- Updated configuration files to clean up and reorganize settings, enhancing clarity and maintainability.
	- Version updates in configuration files to reflect application upgrades.

### CODE REVIEW

File: ct-app/test/components/test_decorators.py [hunk 1]
```diff
@@ -68,13 +68,16 @@ async def test_flagguard(foo_class: FooClass):
...
+            )
+        except asyncio.TimeoutError:
+            pass
+
+        assert foo_class.counter == expected_count
 
-    setup_test(1, 0, 1)
-    setup_test(1.3, 0.5, 2)
+    await setup_test(1, 0, 1)
+    await setup_test(1.3, 0.5, 2)
```
[the remark this answers is not in the document]
Comment (jeandemeusy) [reply ct-app/test/components/test_decorators.py:83]: The proposition requires an extra library, which is not desired. Ignoring recommandation.

File: ct-app/test/test_core.py [hunk 2]
```diff
@@ -69,37 +68,3 @@ async def test_apply_economic_model(core: Core):
 @pytest.mark.asyncio
 async def test_get_fundings(core: Core):
     pytest.skip(f"{inspect.stack()[0][3]} not implemented")
```
[the remark this answers is not in the document]
Comment (jeandemeusy) [reply ct-app/test/test_core.py:70]: No need

File: ct-app/test/components/test_asyncloop.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    assert len(AsyncLoop().tasks) == 2
+
+
+@pytest.mark.asyncio
+@clearAsyncLooopInstance
+async def test_hasRunningTasks():
+    assert not AsyncLoop.hasRunningTasks()
+
+    AsyncLoop.add(foo_awaitable)
+    assert AsyncLoop.hasRunningTasks()
```
[the remark this answers is not in the document]
Comment (jeandemeusy) [reply ct-app/test/components/test_asyncloop.py:52]: Already fixed

File: ct-app/run.sh [hunk 4]
```diff
@@ -0,0 +1,27 @@
+export $(grep -v '^#' .env | xargs)
+
+env=${1:staging}
+count=${2:1}
+
+healthyz() { echo $(curl -s -o /dev/null -w "%{http_code}" "${env}/healthyz"); }
```
[the remark this answers is not in the document]
Comment (jeandemeusy) [reply ct-app/run.sh:6]: Ignoring fue to low confidence.

File: ct-app/run.sh [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+check_deployment() {
+    if [ $(healthyz $(printf $env "blue" "1" $count)) -eq 200 ]; then
+        echo "blue"
+        exit 1
+    elif [ $(healthyz $(printf $env "green" "1" $count)) -eq 200 ]; then
+        echo "green"
+        exit 1
+    fi
```
[the remark this answers is not in the document]
Comment (jeandemeusy) [reply ct-app/run.sh:15]: ignoring as not working.

File: ct-app/run.sh [hunk 6]
```diff
@@ -0,0 +1,27 @@
+export $(grep -v '^#' .env | xargs)
```
[the remark this answers is not in the document]
Comment (jeandemeusy) [reply ct-app/run.sh:2]: Not working. Ignoring.

File: ct-app/run.sh [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+}
+
+deployment=$(check_deployment $HOST_FORMAT $env)
+
+
+# Node parameters
+for i in $(seq 1 $count); do
+    export NODE_ADDRESS_${i}=$(printf $HOST_FORMAT $deployment $i $env)
+    export NODE_KEY_${i}=$TOKEN
+done
```
[the remark this answers is not in the document]
Comment (jeandemeusy) [reply ct-app/run.sh:27]: not necessary. ignoring.

File: ct-app/core/model/peer.py [hunk 8]
```diff
@@ -122,9 +123,13 @@ async def message_delay(self) -> float:
         count = await self.yearly_message_count.get()
 
         if count is not None and count > 0:
-            return (365 * 24 * 60 * 60) / count
+            value = (365 * 24 * 60 * 60) / count
```
Comment (Teebor-Choka) [inline ct-app/core/model/peer.py:127]: Magic variable that breaks on the skip year:
```suggestion
 SECONDS_IN_A_NON_LEAP_YEAR = 365 * 24 * 60 * 60

...

            value = SECONDS_IN_A_NON_LEAP_YEAR / count
```

**Review by Teebor-Choka — APPROVED** [review]

**Comment by Teebor-Choka** [comment]
A test would be nice. end to end test.

**Pull request merged**